### DISCUSSION TITLE
Copilot in Visual Studio after network absence.

### DISCUSSION THREAD

**Discussion by john-preston** [community/community#55915] [Copilot Conversations]
### Select Topic Area

Question

### Body

In case I launch Visual Studio with GitHub Copilot extension without an internet access at the moment, there is no way to restart / try again / turn it on after that even if the network is already available. Only restarting Visual Studio helps and after restart it works as expected. But I'd like it to detect the network state change and reconnect automatically. Or at least allow restarting it without restarting the IDE.

Right now if I launch Visual Studio without internet access and get the connection later I get this log:
```
GitHub Copilot 1.82.1.0 (v1.82.1.0@10c323cc0)
[DEBUG] [agent] [2023-05-19T04:38:25.963Z] Agent service starting
[DEBUG] [agent] [2023-05-19T04:38:26.057Z] Telemetry initialized
[Error] An error occurred for 'SetInitialAuthStatusAsync': StreamJsonRpc.RemoteInvocationException: Request checkStatus failed with message: getaddrinfo ENOTFOUND api.github.com
   at StreamJsonRpc.JsonRpc.<InvokeCoreAsync>d__145`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotLanguageClient.<SetInitialAuthStatusAsync>d__64.MoveNext()
GitHub Copilot Agent v1.82.0
```
And I get this menu:

![image](https://github.com/community/community/assets/17900494/8b6cdf49-47f6-43b7-af14-7b50f7821cac)

In this menu the button "Sign In" does nothing at all, "Learn More..." leads to https://copilot.github.com and nothing can be done until Visual Studio is restarted.

**Comment by htanwar922** [comment]
I second that..

**Reply by ayim** [reply]
@htanwar922 which version of Visual Studio and the GitHub Copilot extension did you have when running into this issue?

If you're still experiencing issues when reading this message, please Report a Problem via the VS Feedback Tool so we can collect more diagnostics to root cause the issue.

![image](https://github.com/community/community/assets/10951869/113e23cd-6c2b-441f-b77a-16c5439d98c7)

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio
